force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -162,10 +191,11 @@ public with sharing class ClassesObjectsMethods {
      * @return A integer between 50 and 100.
      */
     public static Integer generateRandomNumber() {
+        Integer result;
         // Use Math class to generate number between 50 and 100
-
+        result = Integer.valueOf(Math.random() * (100 - 50 + 1)) + 50;
         // Return the random integer
-        return null; // Replace null with the variable you used to store the result
+        return result; 
```

Comment:
just return the value directly. I think your random value is only not going to return  the number 50.